docs/howto/operator/custom-operator.rst

```diff
@@ -0,0 +1,177 @@
+    under the License.
+
+
+Create Custom Operator
+=======================
+
+
+Airflow allows you to create new operators to suit the requirements of you or your team. 
+The extensibility is one of the many reasons which makes Apache Airflow powerful. 
+
+You can create any operator you want by extending the :class:`airflow.models.baseoperator.BaseOperator`
+
+There are two methods that you need to override in derived class:
+
+* Constructor - Define the parameters required for the operator. You only need to specify the arguments specific to your operator.
+
+* Execute - The code to execute when the runner calls the operator. The method contains the 
+  airflow context as a parameter that can be used to read config values.
+
+Let's implement an example ``HelloOperator``:
+
+.. code::  python
+
+        class HelloOperator(BaseOperator):
+
+            def __init__(
+                    self,
+                    name: str,
+                    *args, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                self.name = name
+
+            def execute(self, context):
+                message = "Hello {}".format(name)
+                print(message)
+                return message
+
+The above operator can now be used as follows:
+
+.. code:: python
+
+    hello_task = HelloOperator(task_id='', dag=dag, name='foo_bar')
+
+Hooks
+^^^^^
+Hooks act as an interface to communicate with the external shared resources in a DAG.
+For example, multiple tasks in a DAG can require access to a MySQL database. Instead of
+creating a connection per task, you can retrieve a connection from the hook and utilize it.
+Hook also helps to avoid storing connection auth parameters in a DAG. 
+See :doc:`../connection/index` for how to create and manage connections.
+
+Let's extend our previous example to fetch name from MySQL:
+
+.. code:: python
+
+    class HelloDBOperator(BaseOperator):
+
+            def __init__(
+                    self,
+                    name: str,
+                    conn_id: str,
+                    database: str,
+                    *args, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                self.name = name
+                self.conn_id = conn_id
+                self.database = database
+
+            def execute(self, context):
+                hook = MySqlHook(mysql_conn_id=self.mysql_conn_id,
+                         schema=self.database)
+                sql = "select name from user"
+                result = hook.get_first(sql)
+                message = "Hello {}".format(result['name'])
+                print(message)
+                return message
+
+When the operator invokes the query on the hook object, a new connection gets created if it doesn't exist. 
+The hook retrieves the auth parameters such as username and password from Airflow
+backend and passes the params to the ``get_connection`` method from ``BaseHook``. 
```

Comment:
Can you add link to method documentation?
https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing